docs/src/main/tut/foldable.md

```diff
@@ -86,5 +90,35 @@ as opposed to
 def foldRight[A, B](fa: F[A], z: B)(f: (A, B) => B): B
 ```
  
-which someone familiar with the `foldRight` from the collections in Scala's standard
-library might expect. 
+which someone familiar with the `foldRight` from the collections in
+Scala's standard library might expect. This will prevent operations
+which are lazy in their right hand argument to traverse the entire
+structure unnecessarily. For example, if you have:
+
+```tut
+val allFalse = Stream.continually(false)
+```
+
+which is an infinite stream of `false` values, and if you wanted to
+reduce this to a single false value using the logical and (`&&`). You
+intuitively know that the result of this operation should be
+`false`. It is not necessary to consider the entire stream in order to
+determine this result, you only need to consider the first
+value. Using `foldRight` from the standard library *will* try to
+consider the entire stream, and thus will eventually cause a stack
+overflow:
+
+```tut
+try {
+  allFalse.foldRight(true)(_ && _)
+} catch {
+  case e:StackOverflowError => println(e)
+}
+```
+
+With the lazy `foldRight` on `Foldable`, the calculation terminates
+after looking at only one value:
+
+```tut
+Foldable[Stream].foldRight(allFalse, Eval.now(true))((a,b) => Eval.now(a && b.value)).value
```

Comment:
I think that `(a,b) => Eval.now(a && b.value)` would need to be something like `(a,b) => if (a) b else Eval.now(false)` in order to be stack safe if there were say 10k `true` values prepending the continually `false` values. So it might be a good idea to avoid calling `.value` until the very end, since that's what you usually want to do with `Eval` to achieve stack safety.

Having said that, I just gave it a try and it looks like the Cats `foldRight` implementation for `Stream` isn't actually stack safe. I'll create a separate issue for that.